### DISCUSSION TITLE
Fill the PkgRef of a vulnerability when scanning container images, filesystem, etc

### DISCUSSION THREAD

**Discussion by pablogalegoc** [aquasecurity/trivy#5326] [Ideas]
### Description

I've noticed this comment in the code when trying to find why the vulnerabilities in a report don't have a pURL of the package affected:
https://github.com/aquasecurity/trivy/blob/74dbd8a1fd5a67ccf63e5039893d5c3a9f3536e0/pkg/types/vulnerability.go#L22-L27

Could this decision be reconsidered? It would be nice if a standard scan* would fill the `PkgRef` of the vulnerability, ideally with a pURL, so it is easier to link with VEX statements.

\* I mean something like:
```
trivy -q i --scanners vuln library/redis:7.2.1-bookworm -f json
```

### Target

Container Image

### Scanner

Vulnerability

**Comment by knqyf263** [comment]
Yeah, we're planning to do that.

**Reply by pablogalegoc** [reply]
Awesome, thanks!

**Comment by mpermar** [comment]
@knqyf263 does it sound good if we (Bitnami) contribute a PR for this feature? Probably we should not even ask but I want to make sure you guys don't have any strong opinions on the implementation or are already doing it and waiting for just being pushed.

**Reply by knqyf263** [reply]
Yeah, it will be appreciated! I don't have a concrete idea yet, just wondering if we should add `PURL` to `Package` or `Identifiers.PURL` to `Package` so that we can have `Identifiers.CPE` or other IDs` in the future. If you already have an idea, I'd like to hear that.

**Comment by mpermar** [comment]
@juan131 is going to take a look at this. We do rely purl in our SBOMs, scans and VEX statements but I concur that it makes sense to support other package identification formats.

**Reply by knqyf263** [reply]
Thanks!

**Comment by juan131** [comment]
@pablogalegoc @mpermar @knqyf263 I just created this draft PR with a possible implementation for this: https://github.com/aquasecurity/trivy/pull/5439

**Reply by juan131** [reply]
@knqyf263 please take a look to the above mentioned PR when you find time for it.
